### ISSUE TITLE
Add config file support

### ISSUE THREAD

**Issue by ibnesayeed** [oduwsdl/memgator#2]
Passing command line arguments every time may be tedious, loading config from a file will be good to have.

**Comment by machawk1** [comment]
Would YAML be the best format for this?

**Comment by ibnesayeed** [comment]
YAML would certainly be a very friendly format. However, Go does not have YAML parser built-in, but there is a very good third-party package for that. I did not use that to avoid external dependencies, but it looks like it might be worth including that. If we do so, it would possibly be better to also allow YAML in archive listing. YAML will reduce the chances of trailing comma issue in JSON, especially when copy paste of objects happen. Since JSON is a subset of YAML, existing JSON format would also work.

**Comment by machawk1** [comment]
Is https://github.com/go-yaml/yaml the "very good third party package"?

**Comment by ibnesayeed** [comment]
Yes, that's the one I was referring to.

**Comment by ikreymer** [comment]
+1

**Comment by galsondor** [comment]
If it is not too late, I suggest JSON, which is part of the standard go library.  Two reasons for this:

* Users of memgator are likely to already know JSON,
* Third-party dependencies can be a real long-term maintenance headache&mdash;they are best minimized.

**Comment by ibnesayeed** [comment]
No, it's not too late @galsondor. However there are a few arguments in favor of YAML.

* JSON is almost a subset of YAML that means if someone chooses to use JSON, it will work.
* YAML doesn't have trailing comma issue that is very annoying, especially when one has to copy-paste blocks.
* YAML allows comments which JSON does not.
* YAML files will look more familiar to users of other config file formats such as INI.
* The third party YAML Go library is very popular and quite stable.

**Comment by galsondor** [comment]
@ibnesayeed A popular third-party library is a fairly safe choice and the lack of comments is a big JSON headache.  So, it sounds like YAML with lots of explanatory comments for those who are not YAML experts.
